File: app/src/main/java/es/etologic/mahjongscoring2/domain/use_cases/GetAllPlayersUseCase.java
package es.etologic.mahjongscoring2.domain.use_cases;

import java.util.List;

import javax.inject.Inject;

import es.etologic.mahjongscoring2.data.repositories.PlayersRepository;
import es.etologic.mahjongscoring2.domain.model.Player;
import io.reactivex.Single;

public class GetAllPlayersUseCase {

    private PlayersRepository playersRepository;

    @Inject
    public GetAllPlayersUseCase(PlayersRepository playersRepository) { this.playersRepository = playersRepository; }

    public Single<List<Player>> getAll() {
        return playersRepository.getAllPlayers();
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/domain/use_cases/GetAllPlayersUseCase.kt
package es.etologic.mahjongscoring2.domain.use_cases

import es.etologic.mahjongscoring2.data.repositories.PlayersRepository
import es.etologic.mahjongscoring2.domain.model.Player
import io.reactivex.Single
import javax.inject.Inject

class GetAllPlayersUseCase @Inject
constructor(private val playersRepository: PlayersRepository) {
    
    internal fun getAll(): Single<List<Player>> = playersRepository.getAllPlayers()
}
